tests/e2e/taqueria-plugin-flextesa.spec.ts

```diff
@@ -235,4 +237,231 @@ describe("E2E Testing for taqueria flextesa plugin",  () => {
+
+            // 2. Run taq typecheck hello-tacos-ill-typed.tz
+            const {stdout, stderr} = await exec(`taq typecheck ${dockerName} hello-tacos-ill-typed.tz`, {cwd: `./${taqueriaProjectPath}`})
+
+            // 3. Verify that output includes a table and an error message
+            expect(stdout).toBe(contents.typeError)
+            expect(stderr).toContain("Type nat is not compatible with type string");
+
+        } catch(error) {
+            throw new Error (`error: ${error}`);
+        }
+    })
+
+    test('Verify that taqueria simulator task can simulate one contract using simulate [sourceFile] command', async () => {
+        try {
+            // 1. Copy contract from data folder to taqueria project folder
+            execSync(`cp e2e/data/hello-tacos.tz ${taqueriaProjectPath}/contracts`);
+
+            // 2. Run taq simulate hello-tacos.tz
+            const {stdout, stderr} = await exec(`taq simulate ${dockerName} hello-tacos.tz --storage '5' --input '2'`, {cwd: `./${taqueriaProjectPath}`});
+
+            // 3. Verify that it's valid and contains no errors
+            expect(stdout).toBe(contents.oneRowTableSimulateResult);
+            expect(stderr).toBe("");
+
+        } catch(error) {
+            throw new Error (`error: ${error}`);
+        }
+    });
+
+    test('Verify that taqueria simulator task will display proper message if user tries to simulate contract that does not exist', async () => {
+        try {
+            // 1. Run taq simulate ${contractName} for contract that does not exist
+            const {stdout, stderr} = await exec(`taq simulate ${dockerName} test.tz --storage '5' --input '2'`, {cwd: `./${taqueriaProjectPath}`})
+
+            // 2. Verify that output includes a table and an error message
+            expect(stdout).toBe(contents.nonExistent)
+            expect(stderr).toContain("Does not exist");
+
+        } catch(error) {
+            throw new Error (`error: ${error}`);
+        }
+    });
+
+    test("Verify that taqueria simulator task emits logic error (runtime exception to be thrown)", async () => {
+        try {
+            // 1. Copy contract from data folder to taqueria project folder
+            execSync(`cp e2e/data/hello-tacos.tz ${taqueriaProjectPath}/contracts`)
+
+            // 2. Run taq simulate hello-tacos.tz
+            const {stdout, stderr} = await exec(`taq simulate ${dockerName} hello-tacos.tz --storage '5' --input '10'`, {cwd: `./${taqueriaProjectPath}`})
+
+            // 3. Verify that output includes a table and an error message
+            expect(stdout).toBe(contents.runtimeError)
+            expect(stderr).toContain("NOT_ENOUGH_TACOS");
+
+        } catch(error) {
+            throw new Error (`error: ${error}`);
+        }
+    })
+
+    test("Verify that taqueria simulator task emits parameter type error (supplying string instead of nat)", async () => {
```

Comment:
The title says nat but the error is specifying int, is the actual restriction to natural numbers?

Reply:
That's an error emitted by Tezos-client's simulation command. But yes, the actual restriction is to natural integer (no negative numbers)